build-system/tasks/dep-check.js

```diff
@@ -204,9 +203,12 @@ function getGraph(entryModule) {
  */
 function getEntryModule(extensionFolder) {
   var extension = path.basename(extensionFolder);
-  var versions = extensionsVersions[extension] || ['0.1'];
-  return versions.map(v =>
-      `${extensionFolder}/${v}/${extension}.js`);
+  return fs.readdirAsync(extensionFolder).then(dirItems => {
```

Comment:
is it possible that we name deprecated version folder as `0.1_depr` for example so that it's easy to tell one version is deprecated. Also I don't think it's still necessary to run dep_check to deprecated version.